alacritty/src/display/content.rs

```diff
@@ -478,31 +481,33 @@ impl DerefMut for RegexMatches {
     }
 }
 
-/// Visible regex match tracking.
+/// Visible hint match with current index.
 #[derive(Default)]
-struct Regex<'a> {
+struct TrackedHintMatches<'a> {
     /// All visible matches.
-    matches: Cow<'a, RegexMatches>,
+    matches: Cow<'a, [HintMatchLocation]>,
 
     /// Index of the last match checked.
     index: usize,
 }
 
-impl<'a> Regex<'a> {
-    /// Create a new renderable regex iterator.
+impl<'a> TrackedHintMatches<'a> {
+    /// Create a new renderable iterator.
     fn new<T>(term: &Term<T>, dfas: &RegexSearch) -> Self {
-        let matches = Cow::Owned(RegexMatches::new(term, dfas));
-        Self { index: 0, matches }
+        let matches = RegexMatches::new(term, dfas);
+        let matches =
+            matches.0.into_iter().map(|bounds| HintMatchLocation::Regex { bounds }).collect();
+        Self { index: 0, matches: Cow::Owned(matches) }
     }
 
-    /// Advance the regex tracker to the next point.
+    /// Advance the tracker to the next point.
     ///
-    /// This will return `true` if the point passed is part of a regex match.
+    /// This will return `true` if the point passed is part of a hint match.
     fn advance(&mut self, point: Point) -> bool {
-        while let Some(regex_match) = self.matches.get(self.index) {
-            if regex_match.start() > &point {
+        while let Some(matched) = self.matches.get(self.index) {
```

Comment:
```suggestion
        while let Some(match) = self.matches.get(self.index) {
```

Same thing again, `matched` as a noun doesn't seem like a great variable name when just "match" is available.

Reply:
There's a keyword `match`. I use `matched` everywhere just to workaround it.

Reply:
Ah right, that's why I was using `regex_match`, really don't like `matched` though it's not a great name.